### PR TITLE
Reform the benchmark rules.

### PR DESCRIPTION
**Pull request by xingyu-long**
We provide the benchmarks by sorted batch_size, and also provide GPU benchmark with maximum batch_size for current model. For `cifar10_cnn_benchmark_test`, we used the small epochs to get the results quickly.

### CODE REVIEW

File: tensorflow/python/keras/benchmarks/keras_examples_benchmarks/cifar10_cnn_benchmark_test.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def __init__(self):
     (self.x_train, self.y_train), _ = tf.keras.datasets.cifar10.load_data()
     self.x_train = self.x_train.astype('float32') / 255
     self.y_train = tf.keras.utils.to_categorical(self.y_train, self.num_classes)
-    self.epochs = 25
+    self.epochs = 5
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/keras_examples_benchmarks/cifar10_cnn_benchmark_test.py:34]: IMO, we can keep it as long as it's not timeout.

File: tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_conv_benchmark_test.py [hunk 2]
```diff
@@ -96,21 +96,16 @@ def benchmark_conv_mnist_bs_256(self):
...
-  def benchmark_conv_mnist_bs_256_gpu_2(self):
-    """Measure performance with batch_size=256, run_iters=3, gpu=2 and
-
-    distribution_strategy='mirrored'
-    """
-    batch_size = 256
+  def benchmark_conv_mnist_bs_512(self):
+    """Measure performance with batch_size=512 and run_iters=3."""
+    batch_size = 512
     run_iters = 3
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_conv_benchmark_test.py:102]: Can we use a larger `run_iters`, if it doesn't take too long for one iter? like 5?

File: tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_conv_benchmark_test.py [hunk 3]
```diff
@@ -119,8 +114,11 @@ def benchmark_conv_mnist_bs_256_gpu_2(self):
...
 
-  def benchmark_conv_mnist_bs_512(self):
-    """Measure performance with batch_size=512 and run_iters=3."""
+  def benchmark_conv_mnist_bs_512_gpu_2(self):
+    """Measure performance with batch_size=512, run_iters=3, gpu=2 and
+
+    distribution_strategy='mirrored'
+    """
     batch_size = 512
     run_iters = 3
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_conv_benchmark_test.py:123]: ditto as above.

File: tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_hierarchical_rnn_benchmark_test.py [hunk 4]
```diff
@@ -78,32 +78,27 @@ def benchmark_hrnn_mnist_bs_256(self):
...
         iters=run_iters, wall_time=wall_time, metrics=metrics, extras=extras)
 
-  def benchmark_hrnn_mnist_bs_512(self):
-    """Measure performance with batch_size=512 and run_iters=5."""
-    batch_size = 512
-    run_iters = 5
+  def benchmark_hrnn_mnist_bs_1024(self):
+    """Measure performance with batch_size=1024 and run_iters=3."""
+    batch_size = 1024
+    run_iters = 3
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_hierarchical_rnn_benchmark_test.py:101]: A large batch size means running faster, so we can keep iters as 5

File: tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_hierarchical_rnn_benchmark_test.py [hunk 5]
```diff
@@ -117,16 +112,21 @@ def benchmark_hrnn_mnist_bs_512(self):
...
-  def benchmark_hrnn_mnist_bs_1024(self):
-    """Measure performance with batch_size=1024 and run_iters=3."""
+  def benchmark_hrnn_mnist_bs_1024_gpu_2(self):
+    """Measure performance with batch_size=1024, run_iters=4, gpu=2 and
+
+    distribution_strategy='mirrored'
+    """
     batch_size = 1024
-    run_iters = 3
+    run_iters = 4
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_hierarchical_rnn_benchmark_test.py:121]: Ditto as above: let's keep the `run_iters` the same for the same batch_size and the same model.

File: tensorflow/python/keras/benchmarks/keras_examples_benchmarks/reuters_mlp_benchmark_test.py [hunk 6]
```diff
@@ -119,16 +114,21 @@ def benchmark_mlp_reuters_bs_256(self):
...
-  def benchmark_mlp_reuters_bs_512(self):
-    """Measure performance with batch_size=512 and run_iters=4."""
+  def benchmark_mlp_reuters_bs_512_gpu_2(self):
+    """Measure performance with batch_size=512, run_iters=2, gpu=2 and
+
+    distribution_strategy='mirrored'
+    """
     batch_size = 512
-    run_iters = 4
+    run_iters = 2
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/keras_examples_benchmarks/reuters_mlp_benchmark_test.py:123]: ditto as above.

File: tensorflow/python/keras/benchmarks/keras_examples_benchmarks/reuters_mlp_benchmark_test.py [hunk 7]
```diff
@@ -78,21 +78,16 @@ def benchmark_mlp_reuters_bs_128(self):
...
-    """Measure performance with batch_size=128, run_iters=2, gpu=3 and
-
-    distribution_strategy='mirrored'
-    """
-    batch_size = 128
-    run_iters = 2
+  def benchmark_mlp_reuters_bs_256(self):
+    """Measure performance with batch_size=256 and run_iters=3."""
+    batch_size = 256
+    run_iters = 3
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/keras_examples_benchmarks/reuters_mlp_benchmark_test.py:84]: ditto as above: a larger run_iters if it doesn't take too long.

**Review by yhliang2018 — COMMENTED** [review]
Thanks for the PR!

**Comment by xingyu-long** [comment]
@yhliang2018 I think this PR should be approved.

**Review by yhliang2018 — APPROVED** [review]
Could you also update the benchmark numbers in the reference table? Probably in next PR? Otherwise, there is a mismatch between benchmarks and the numbers.

**Pull request merged**